src/routes/users.rs

```diff
@@ -544,3 +561,209 @@ pub async fn user_notifications(
+            ));
+        }
+
+        let mut transaction = pool.begin().await?;
+
+        if let Some(email_setting) = new_settings.public_email {
+            sqlx::query!(
+                "
+                UPDATE user_settings
+                SET public_email = $1
+                WHERE user_id = $2
+                ",
+                email_setting,
+                id as crate::database::models::ids::UserId
+            )
+            .execute(&mut *transaction)
+            .await?;
+        }
+
+        if let Some(github_setting) = new_settings.public_github {
+            sqlx::query!(
+                "
+                UPDATE user_settings
+                SET public_github = $1
+                WHERE user_id = $2
+                ",
+                github_setting,
+                id as crate::database::models::ids::UserId
+            )
+            .execute(&mut *transaction)
+            .await?;
+        }
+
+        if let Some(theme_setting) = &new_settings.theme {
+            sqlx::query!(
+                "
+                UPDATE user_settings
+                SET theme = $1
+                WHERE user_id = $2
+                ",
+                theme_setting.as_str(),
+                id as crate::database::models::ids::UserId
+            )
+            .execute(&mut *transaction)
+            .await?;
+        }
+
+        transaction.commit().await?;
+        Ok(HttpResponse::NoContent().body(""))
+    } else {
+        Ok(HttpResponse::NotFound().body(""))
+    }
+}
+
+#[patch("settings")]
+pub async fn user_settings_edit_from_header(
+    req: HttpRequest,
+    pool: web::Data<PgPool>,
+    new_settings: web::Json<NewSettings>,
+) -> Result<HttpResponse, ApiError> {
+    let user = get_user_from_headers(req.headers(), &**pool).await?;
+
+    let mut transaction = pool.begin().await?;
+
+    if let Some(email_setting) = new_settings.public_email {
+        sqlx::query!(
+            "
+            UPDATE user_settings
+            SET public_email = $1
+            WHERE user_id = $2
+            ",
+            email_setting,
+            user.id.0 as i64
+        )
+        .execute(&mut *transaction)
+        .await?;
+    }
+
+    if let Some(github_setting) = new_settings.public_github {
+        sqlx::query!(
+            "
+            UPDATE user_settings
+            SET public_github = $1
+            WHERE user_id = $2
+            ",
+            github_setting,
+            user.id.0 as i64
+        )
+        .execute(&mut *transaction)
+        .await?;
+    }
+
+    if let Some(theme_setting) = &new_settings.theme {
+        sqlx::query!(
+            "
+            UPDATE user_settings
+            SET theme = $1
+            WHERE user_id = $2
+            ",
+            theme_setting.as_str(),
+            user.id.0 as i64
+        )
+        .execute(&mut *transaction)
+        .await?;
+    }
+
+    transaction.commit().await?;
+
+    Ok(HttpResponse::NoContent().body(""))
+}
```

Comment:
I don't see the use for these routes. The database already gets user settings with the user. Why not return settings when the user is requested in the first place by token & by ID (and just hide it if not authorized)?

Also- I would put the user settings inside of the user edit route. No need for a new route